### PR TITLE
overload for bigtable.getInstances() 

### PR DESCRIPTION
**Pull request by laljikanjareeya**
while accessing `bigtable.getInstances()` in `async` function, it was giving error of return type.
added overload and related document changes.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -610,16 +657,34 @@ export class Bigtable {
...
    */
-  getInstances(gaxOptions?, callback?) {
-    if (is.function(gaxOptions)) {
-      callback = gaxOptions;
-      gaxOptions = {};
+  getInstances(
+    gaxOptionsOrCallback?: GetInstancesRequest | GetInstancesCallback,
+    callback?: GetInstancesCallback
+  ): void | Promise<GetInstancesResponse> {
+    if (is.function(gaxOptionsOrCallback)) {
```
Comment (AVaksman) [inline src/index.ts]: here rather introduce a `gaxOptions` variable (not `gaxOptionsOrCallback`)
Something like
```ts
const gaxOptions = 
  typeof gaxOptionsOrCallback === 'object' ? optionsOrCallback : {};
callback =
  typeof gaxOptionsOrCallback === 'function' ? optionsOrCallback : callback;
```

File: src/index.ts [hunk 2]
```diff
@@ -631,11 +696,11 @@ export class Bigtable {
         client: 'BigtableInstanceAdminClient',
         method: 'listInstances',
         reqOpts,
-        gaxOpts: gaxOptions,
+        gaxOpts: gaxOptionsOrCallback,
```
Comment (AVaksman) [inline src/index.ts]: Once you introduce `gaxOptions` variable as per comment above, this change will not be needed.

File: src/index.ts [hunk 3]
```diff
@@ -588,16 +609,42 @@ export class Bigtable {
...
+   * @callback GetInstancesCallback
+   * @param {?Error} err Request error, if any.
+   * @param {Instance[]} instances Array of {@link Instance} instances.
+   * @param {object} apiResponse The full API response.
+   */
   /**
    * Get Instance objects for all of your Cloud Bigtable instances.
    *
-   * @param {object} [gaxOptions] Request configuration options, outlined here:
+   * @param {GetInstancesRequest} [gaxOptions] Request configuration options, outlined here:
```
Comment (AVaksman) [inline src/index.ts]: This should probably be `object` again?
Also I believe we should fix the link below to 
https://googleapis.github.io/gax-nodejs/classes/CallSettings.html

File: src/index.ts [hunk 4]
```diff
@@ -588,16 +609,42 @@ export class Bigtable {
...
+  getInstances(callback: GetInstancesCallback): void;
+  getInstances(
+    gaxOptions: GetInstancesRequest,
+    callback: GetInstancesCallback
+  ): void;
+  getInstances(gaxOptions?: GetInstancesRequest): Promise<GetInstancesResponse>;
+  /**
+   * Query object for listing instances.
+   *
+   * @typedef {object} GetInstancesRequest
```
Comment (AVaksman) [inline src/index.ts]: `GetInstancesRequest` => `CallOptions`

File: src/index.ts [hunk 5]
```diff
@@ -588,16 +609,42 @@ export class Bigtable {
     );
   }
 
+  getInstances(callback: GetInstancesCallback): void;
+  getInstances(
+    gaxOptions: GetInstancesRequest,
```
Comment (AVaksman) [inline src/index.ts]: `GetInstancesRequest` => `CallOptions` here and everywhere below

**Review by AVaksman — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (laljikanjareeya) [reply src/index.ts]: Done!

**Review by AVaksman — APPROVED** [review]

**Pull request closed without merging**